nft-client/src/main.rs

```diff
@@ -499,55 +575,59 @@ async fn main() -> anyhow::Result<()> {
                             )
                         })?
                         .0;
-                    cis1::Receiver::Contract(convert_contract_address(&address), receive_name)
+                    cis2::Receiver::Contract(convert_contract_address(&address), receive_name)
                 }
             };
 
             let transfers = token_ids
                 .iter()
-                .map(|token_id| cis1::Transfer {
+                .map(|token_id| cis2::Transfer {
                     token_id: token_id.clone(),
-                    amount:   1,
+                    amount:   cis2::TokenAmount::from(1),
                     from:     convert_address(from.0.clone()),
                     to:       to.clone(),
                     data:     to_func_data.clone(),
                 })
                 .collect();
 
-            let parameter = cis1::TransferParams(transfers);
-            let bytes = concordium_contracts_common::to_bytes(&parameter);
-
-            let payload = transactions::UpdateContractPayload {
-                amount:       common::types::Amount::from(0),
-                address:      contract.0,
-                receive_name: smart_contracts::ReceiveName::try_from(format!(
-                    "{}.transfer",
-                    NFT_CONTRACT_NAME
-                ))
-                .map_err(|e| anyhow!("Failed to parse receive name {}", e))?,
-                message:      smart_contracts::Parameter::from(bytes),
-            };
-
-            let transaction_payload = transactions::Payload::Update { payload };
-            eprintln!(
-                "Transferring tokens {:?} from {} to {:?}",
-                token_ids, from, to
-            );
             // The GRPC client for Concordium node.
             let mut grpc_client = Client::connect(app.endpoint, "rpcadmin".to_string())
                 .await
                 .context("Failed to connect to Node GRPC")?;
 
+            let cis2_contract =
+                cis2::Cis2Contract::new(grpc_client.clone(), contract.0, "CIS2-NFT".to_string());
+            let next_account_nonce = grpc_client
+                .get_next_account_nonce(&account_data.address)
+                .await?;
+            ensure!(
+                next_account_nonce.all_final,
+                "There are unfinalized transactions. Transaction nonce is not reliable enough."
+            );
+            let expiry = common::types::TransactionTime::from_seconds(
+                (chrono::Utc::now().timestamp() + 300) as u64,
+            );
             // Measured by experiments, will likely break if contract state becomes large:
-            let estimated_energy = 300 * u64::try_from(token_ids.len())? + 2000;
+            let estimated_energy = 300 * u64::try_from(token_ids.len())? + 5000;
```

Comment:
I'm not a fan of this hardcoding. Would it be too much to have this as a command-line parameter.